app/workers/mail_notification_job.rb

```diff
@@ -40,6 +42,7 @@ def notify
     # Since we cannot recover from this error we catch it and move on.
     Rails.logger.error "Cannot deliver notification (#{self.inspect})
                         as required record was not found: #{e}".squish
+    raise e if raise_exceptions
```

Comment:
Suppressing of exceptions are implemented only for ActiveRecord::RecordNotFound, all other exceptions mark a job as failed, for example: https://github.com/barhot/openproject/blob/feature/allow-failing-notification-jobs/spec/workers/mail_notification_jobs_spec.rb#L95

Usually it's OK, if a record is not found then the job is odd, except the cases when a record exists but the exception happens for some reason. We met such appearance when the DB server was overloaded, probably fallback server responded that time but was not synchronized at that moment. In that case rerunning the failed job might be a good idea.